webserver/webserver/src/main/java/io/helidon/webserver/ServerRequest.java

```diff
@@ -136,4 +141,35 @@
      * @return the tracer associated, or {@link io.opentracing.util.GlobalTracer#get()}
      */
     Tracer tracer();
+
+    /**
+     * Absolute URI of the incoming request, including query parameters and fragment.
+     * The host and port are obtained from the interface this server listens on ({@code host} header is not used).
+     *
+     * @return the URI of incoming request
+     */
+    default URI absoluteUri() {
```

Comment:
Maybe this could be a lazily initialized field on the implementation, wdyt ?

Reply:
Updated to use Lazy and added a test.